### DISCUSSION TITLE
syuilo氏の態度と「ダウンストリーム」を顧みない姿勢について

### DISCUSSION THREAD

**Discussion by KisaragiEffective** [misskey-dev/misskey#12069] [General]
おおよそ問題とされている点は取り除いたと思うので、回答を期待します

# 改定履歴
1. フォークを「ダウンストリーム」から削除
2. CoCに抵触した表現を削除

## 用語の定義
* ダウンストリーム: 下記いずれかに当てはまる者を指す。
    1. サーバー管理者
    3. エンドユーザー
    4. ~~フォーク先~~ (edit: 削除)
    5. サードパーティークライアントの開発者

## 問題点
* コミットログを見てもなんの変更なのかわからずコミットを精査するときに**非常に手間**
    * 暗黙の文脈は他人が理解できるとは限りません。理解できたとしても時間がかかることは必然です。
        * 例: [直近のコミット: "update test"](https://github.com/misskey-dev/misskey/commit/5b244cc6eb609ec5078dcb3898f14d9004465899)
            * -> タイムラインのE2Eテストに件数の上限を設定している
* ダウンストリームにとって破壊的変更となる変更履歴で何がどう変わったのか、ダウンストリームによって手動で行う必要があるマイグレーションをどうするべきなのかといった情報が不十分
* APIが不安定
    * ドキュメントがoutdated
        * 詳細: [misskey-hubの当該issue](https://github.com/misskey-dev/misskey-hub/issues/222)
    * OpenAPI「っぽい」JSONを出力できるものの、規格に厳密に沿う処理系だと解釈できない
        * 詳細: #10632 
    * [megalodonのMisskeyサポートを終了する](https://h3poteto.hatenablog.com/entry/2023/09/25/220706) [^1]
        * > つまりコントリビュートしている人にとっては，APIドキュメントはそんなに重要ではなく，サードパーティー製のアプリケーション開発者も大して大事じゃないだろう
         
            破壊的変更が多いにもかかわらず、ドキュメントが整備されていないと事実上ないに等しいです (disclaimer: サーバーのバージョンによってスキーマが変わるのは一般に非常にしんどいです)

## 疑問点
1. セマンティックコミットにところどころ沿ってるのになんで沿っていないコミットは沿ってないんですか？
   * 例: いちいち上げてると**キリ**がないので[gist](https://gist.github.com/KisaragiEffective/96fdf83852bf5c93382128d52f0f6af3)を見てください
3. 時々gitmojiを使うのをやめてもらえますか？
   * 例: b349d0baf8f5552f185b3d93cbd407d34e65cb9e
   * 例: bcbcaa9c60ce5f8de44678976c113437c7eb4356
   * 例: f5e72f7d3ed05b55406b45f59d8169bfeb353559
4. 今後ダウンストリームによって手動で行う必要があるマイグレーションをどうするのか情報を提供する予定はありますか？
6. 今後misskey-hubのissueを処理するつもりはありますか？
7. misskey-hubが現状の実装と乖離しているのはご存知のとおりだと思いますが、今後misskey-hubにおいてパラメーターの追加/削除/セマンティクスのdocumentingを行うつもりはありますか？
8. ダウンストリームのことを考えたことはありますか？

## 私見
Misskeyの開発者体験は全くと言っていいほど最悪です。これ以上「不必要な」非一貫性が招かれないことを切に願います。

~~また、 (#11154 に限りませんが) 氏の「コードさえ書ければダウンストリームはどうでもいい」と取れるような態度はエンドユーザーの観点から見ると非常に不愉快です。~~
~~元々は氏のプロジェクトであることは**重々**承知ですが、今後は氏のものだけではないことは念頭に入れていただきたいと思います。~~
↑CoCに抵触しない表現が見つからないため一時取り消し

なお、この文章はKisaragiEffectiveによる書面による明示的な許可がない限り`github.com`外へその一部または全部を転載することを禁止します。

[^1]: megalodonはMisskeyに対応したサードパーティークライアント**でした**。現在はサポートを打ち切っていると思われます。

**Comment by m1sk9** [comment]
> セマンティックコミットにところどころ沿ってるのになんで沿っていないコミットは沿ってないんですか？

これについては私も感じていたことで, [Contribution guide](https://github.com/misskey-dev/misskey/blob/de6e15e36d1ddd5b2202b509ab4d04109ce3bd11/CONTRIBUTING.md) で [**セマンティックコミット** (明言はされていませんが) に従うよう案内されていて](https://github.com/misskey-dev/misskey/blob/de6e15e36d1ddd5b2202b509ab4d04109ce3bd11/CONTRIBUTING.md#:~:text=Thank%20you%20for,a%20single%20PR.), Misskey にコミットするコントリビューターの方は従っているのにも関わらずメイン開発者である貴方が従っていないのは何故でしょうか. Git Blame が言うにこのセクションは貴方が編集した部分みたいですけど.

![CleanShot 2023-10-18 at 20 44 51](https://github.com/misskey-dev/misskey/assets/82575685/db9f1176-f1c0-4da1-ac93-82b30ece8829)

コミットハッシュ https://github.com/misskey-dev/misskey/commit/b75184ec8e3436200bacdcd832e3324702553d20 のような2万行にも登る変更を軽率に1つのコミット(PR)に含めたりなどとても misskey.io など大人数が使うOSSとは思えない開発環境で正直驚いています.

**Comment by mattyatea** [comment]
この議論自体、感情的になっていて強い言葉や場に合わない言葉を使っていて、人格をも否定するような文章だとも読み取ることもできます。
そもそも議論として成立していないものに見えますのでまずは落ち着いてから考えたほうが良いのではないか、と感じました。
以上です。

**Comment by fruitriin** [comment]
Effective氏を出禁にしたほうが健全だと思う
なんか普通にブロックできるらしい

https://docs.github.com/ja/communities/maintaining-your-safety-on-github/blocking-a-user-from-your-organization

**Comment by kakkokari-gtyih** [comment]
Lock Conversationしておきましょうかね → **しました**

問題点の指摘は建設的であればむしろすべきですが、自らの感情を持ち込んで議論するのはよろしくないと考えます

**Comment by acid-chicken** [comment]
感情的な部分を除いたとしても fork の面倒も見ろはいささか暴論では
